ipatests/test_integration/test_trust.py

```diff
@@ -344,3 +355,121 @@ def test_trustdomains_found_in_external_rootdomain_trust(self):
 
     def test_remove_external_rootdomain_trust(self):
         self.remove_trust(self.ad)
+
+    # Test for one-way forest trust with shared secret
+
+    def test_establish_forest_trust_with_shared_secret(self):
+        self.configure_dns_and_time(self.ad)
+        tasks.configure_windows_dns_for_trust(self.ad, self.master)
+
+        # this is a workaround for
+        # https://bugzilla.redhat.com/show_bug.cgi?id=1711958
+        self.master.run_command(
+            ['ipa', 'dnsrecord-add', self.master.domain.name,
+             self.srv_gc_record_name,
+             '--srv-rec', self.srv_gc_record_value])
+
+        # create windows side of trust using powershell bindings
+        # to .Net functions
+        ps_cmd = (
+            '[System.DirectoryServices.ActiveDirectory.Forest]'
+            '::getCurrentForest()'
+            '.CreateLocalSideOfTrustRelationship("{}", 1, "{}")'.format(
+                self.master.domain.name, self.shared_secret))
+        self.ad.run_command(['powershell', '-c', ps_cmd])
+
+        # create ipa side of trust
+        tasks.establish_trust_with_ad(
+            self.master, self.ad_domain, shared_secret=self.shared_secret)
+
+    def test_trustdomains_found_in_forest_trust_with_shared_secret(self):
```

Comment:
Should not we make this part of test case test_establish_forest_trust_with_shared_secret as validation task after trust-add?

Reply:
No, I think we should separately check that trust is established and that it is operating, like it is done for other types of trust. Best practice for writing tests is to make one assertion in each test case.

Reply:
What if check is not established ? This will get executed and fail.
IMO one def test_* should cover the complete test case possibly including setup/cleanup.

Reply:
We are doing sanity check of new way for  trust establishments. Test cases here are:
1) trust establishment
2) checking that domains can be retreived
3) checking that users can be retreived

Here trust establishment should not be treated as setup but as a testcase